nitro-protocol/test/contracts/LedgerFinancingApp/requireStateSupported.test.ts

```diff
@@ -0,0 +1,291 @@
+    };
+    const pfStateSignedByBoth: RecoveredVariablePart = {
+      variablePart: getVariablePart(supportproofState),
+      signedBy: BigNumber.from(0b11).toHexString(),
+    };
+
+    advanceOneDay();
+
+    const challengeState: State = {
+      ...baseState,
+      appData: appDataABIEncode(supportproofAppData),
+      turnNum: baseState.turnNum + 1,
+      outcome: computeOutcome({
+        [MAGIC_NATIVE_ASSET_ADDRESS]: {[intermediary]: 506, [merchant]: 494}, // 506 is unfair: should be 505
+      }),
+    };
+    const updatedWithIntermediarySignature: RecoveredVariablePart = {
+      variablePart: getVariablePart(challengeState),
+      signedBy: BigNumber.from(0b01).toHexString(),
+    };
+    await expectRevert(
+      () =>
+        ledgerFinancingApp.requireStateSupported(
+          fixedPart,
+          [pfStateSignedByBoth],
+          updatedWithIntermediarySignature
+        ),
+      'earned<claimed'
+    );
+  });
+
+  it('rejects unilateral unsupported candidates', async () => {
+    // construct challenges with unsupported candidates signed by:
+    // - only the intermediary
+    // - only the merchant
+    // assert failure
+
+    await expectRevert(
+      () => ledgerFinancingApp.requireStateSupported(fixedPart, [], signedByMerchant),
+      '!unanimous; |proof|=0'
+    );
+    await expectRevert(
+      () => ledgerFinancingApp.requireStateSupported(fixedPart, [], signedByIntermediary),
+      '!unanimous; |proof|=0'
+    );
+  });
+
+  it('rejects unilateral support proof states', async () => {
+    // construct challenges where the proof state is signed by
+    //  - only the intermediary
+    //  - only the merchant
+    // assert failure.
+
+    await expectRevert(
+      () =>
+        ledgerFinancingApp.requireStateSupported(
+          fixedPart,
+          [signedByMerchant],
+          signedByIntermediary
+        ),
+      '!unanimous proof state'
+    );
+
+    await expectRevert(
+      () =>
+        ledgerFinancingApp.requireStateSupported(
+          fixedPart,
+          [signedByIntermediary],
+          signedByMerchant
+        ),
+      '!unanimous proof state'
+    );
+  });
+
+  it('rejects too-long proofs', async () => {
+    // construct a challenge with two proof states, assert failure.
+
+    await expectRevert(
+      () =>
+        ledgerFinancingApp.requireStateSupported(
+          fixedPart,
+          [signedByMerchant, signedByIntermediary], // two proof states - should fail
+          signedByIntermediary
+        ),
+      '|proof| > 1'
+    );
+  });
+});
+
+/**
+ * increase blocknumber on provider by 7500.
+ * each day is ~7200= blocks (24*60*60/12)
+ */
+function advanceOneDay() {
+  // note: this is a hacky way to advance so many blocks, and results
+  // in a slower test.
+  // The 'hardhat_mine' method is better, but it causes a different error
+  // in the test.
+  for (let i = 0; i < 7500; i++) {
+    provider.send('evm_mine', []);
```

Comment:
This ought to be speed up your tests https://hardhat.org/hardhat-network/docs/reference#hardhat_mine 

Possibly also useful https://hardhat.org/hardhat-network/docs/reference#evm_setnextblocktimestamp

Reply:
Apologies, just read the code comments!

Reply:
What's the error you see with `hardhat_mine`?

Reply:
Something to do with broken gas estimations - I found some documentation around that issue and root causes, but nothing that related it to `hardhat_mine`.

Reply:
This will be a non-issue with the conversion to BPY.